src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceTest.java

```diff
@@ -74,6 +78,14 @@ void testCreateLinksForGeneralPartnerReturnsSuccess() throws ServiceException {
         assertEquals(expectedUri, sentSubmission.getLinks().get("self"));
     }
 
+    @Test
+    void testExceptionIsThrownWhenFirstAndSecondNationalityIsTheSame() throws ServiceException, MethodArgumentNotValidException, NoSuchMethodException {
+        GeneralPartnerDto dto = createDto();
+        dto.getData().setNationality2(Nationality.BELGIAN);
+        Transaction testTransaction = buildTransaction();
+        assertThrows(MethodArgumentNotValidException.class, () -> { generalPartnerService.createGeneralPartner(testTransaction, dto, REQUEST_ID, USER_ID); });
+    }
+
```

Comment:
also in this file there's a few variables like GeneralPartnerDto and GeneralPartnerResource that are named incorrectly and should start with lowercase letter

Reply:
Decapitalized